pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/AbstractIndexType.java

```diff
@@ -62,6 +70,38 @@ public IndexReaderFactory<IR> getReaderFactory() {
     return _readerFactory;
   }
 
+  public void convertToNewFormat(TableConfig tableConfig, Schema schema) {
+    Map<String, C> deserialize = getConfig(tableConfig, schema);
+    for (Map.Entry<String, C> entry : deserialize.entrySet()) {
+      AtomicBoolean fieldConfigFound = new AtomicBoolean(false);
+      List<FieldConfig> fieldConfigList = Optional.ofNullable(tableConfig.getFieldConfigList())
+          .orElse(List.of())
+          .stream()
+          .map(fc -> {
+            if (fc.getName().equals(entry.getKey())) {
+              fieldConfigFound.set(true);
```

Comment:
It is not recommended to modify external state on a stream. In this case is fine because we know there is only one element that fulfill the condition, but between this and the fact that Optional.stream is not Java 8 compatible, I would suggest to use a for instead of stream